WalletWasabi/Crypto/Extensions.cs

```diff
@@ -37,5 +38,30 @@ public static class Extensions
 				yield return resultSelector(e1.Current, e2.Current, e3.Current);
 			}
 		}
+
+		/// <summary>
+		/// Executes <paramref name="action"/> on each element of the <paramref name="list"/>
+		/// regardless if one throws an exception. Then throws AggregateException
+		/// containing all exceptions if any. Unpacks one level of AggregateException
+		/// thrown from an action.
+		/// </summary>
+		public static void ForEachAggregateExceptions<T>(this IEnumerable<T> list, Action<T> action, string? message = null)
```

Comment:
bikeshedding: rename to `ForEachAggregatingExceptions`, since the exception aggregation is modifying how the foreach proceeds but "aggregate exceptions" could imply the list is expected to contain those